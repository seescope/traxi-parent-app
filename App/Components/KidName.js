import React, { PropTypes } from 'react';
import { Text, StyleSheet } from 'react-native';
import { WHITE, TRANSPARENT } from '../Constants/Colours';
import { isIOS } from '../Utils';

const KidNameStyle = StyleSheet.create({
  text: {
    color: WHITE,
    textAlign: 'center',
    fontWeight: 'normal',
    fontSize: isIOS ? 19 : 14,
    backgroundColor: TRANSPARENT,
  },
});

const KidName = ({ children }) => (
  <Text style={KidNameStyle.text}>{children}</Text>
);

KidName.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
  ]).isRequired,
};

export default KidName;
