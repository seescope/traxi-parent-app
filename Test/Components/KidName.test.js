import React from 'react';
import renderer from 'react-test-renderer';
import KidName from '../../App/Components/KidName';

describe('<KidName>', () => {
  it('renders a kid\'s name', () => {
    const TEST_NAME = 'First Last';
    const tree = renderer.create(<KidName>{TEST_NAME}</KidName>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handles null last name', () => {
    const TEST_NAME = 'First null';
    const tree = renderer.create(<KidName>{TEST_NAME}</KidName>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
