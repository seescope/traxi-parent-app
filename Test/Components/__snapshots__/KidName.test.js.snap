exports[`<KidName> handles null last name 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "backgroundColor": "transparent",
      "color": "#FFF",
      "fontFamily": "Raleway-Bold",
      "fontSize": 19,
      "textAlign": "center",
    }
  }>
  First 
</Text>
`;

exports[`<KidName> renders a kid's name 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "backgroundColor": "transparent",
      "color": "#FFF",
      "fontFamily": "Raleway-Bold",
      "fontSize": 19,
      "textAlign": "center",
    }
  }>
  First Last
</Text>
`;
